### PR TITLE
Added various sections into readme

### LINKED ISSUE #57: Enhance Readme
**Is your feature request related to a problem? Please describe.**
Enhancing Readme to improve overall documentation

**Describe the solution you'd like**
adding clear and concise description of various features of testcontainers.net and various other information to make readme useful for first time usage of project.

**Describe alternatives you've considered**
NA

**Additional context**
Additonally, there are enhancements will be added for contributing.md and overall documentation of project.

#docs

### PR DESCRIPTION
**Pull request by lalitkale**
Sections Added
- About
- Getting Started
- Features [Features section is focused from end-user standpoint. This can serve as our roadmap. We can tick each box as we proceed further for merging all three implementations] - Need review
- Commands- Need review 
- Examples
- Contributing
- Versioning
- Authors 
- License
- Acknowledgements

closes #57

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,138 @@
-# Testcontainers
-[![Build status](https://ci.appveyor.com/api/projects/status/b20um5dc0ikf8cme/branch/master?svg=true)](https://ci.appveyor.com/project/swissarmykirpan/testcontainers-dotnet-ak6r1/branch/master)
+# Testcontainers.NET
+[![Build status](https://ci.appveyor.com/api/projects/status/b20um5dc0ikf8cme/branch/master?svg=true)](https://ci.appveyor.com/project/swissarmykirpan/testcontainers-dotnet-ak6r1/branch/master)[![Coverage Status](https://coveralls.io/repos/github/testcontainers/testcontainers-dotnet/badge.svg?branch=)](https://coveralls.io/github/testcontainers/testcontainers-dotnet?branch=master) <a href="https://www.nuget.org/packages/TestContainers"><img src="https://img.shields.io/nuget/v/TestContainers.svg?style=flat"></a> [![NuGet](https://img.shields.io/nuget/dt/TestContainers.svg)](https://www.nuget.org/packages/TestContainers)
```
Comment (isen-ng) [inline README.md:2]: You can split the badges into new lines in text without it having new lines in the rendered page. This is more readable when the text is not rendered.

```suggestion
[![Build status](https://ci.appveyor.com/api/projects/status/b20um5dc0ikf8cme/branch/master?svg=true)](https://ci.appveyor.com/project/swissarmykirpan/testcontainers-dotnet-ak6r1/branch/master)
[![Coverage Status](https://coveralls.io/repos/github/testcontainers/testcontainers-dotnet/badge.svg?branch=)](https://coveralls.io/github/testcontainers/testcontainers-dotnet?branch=master) <a href="https://www.nuget.org/packages/TestContainers"><img src="https://img.shields.io/nuget/v/TestContainers.svg?style=flat"></a> 
[![NuGet](https://img.shields.io/nuget/dt/TestContainers.svg)](https://www.nuget.org/packages/TestContainers)
```

File: README.md [hunk 2]
```diff
@@ -1,6 +1,138 @@
...
 
-[![Coverage Status](https://coveralls.io/repos/github/testcontainers/testcontainers-dotnet/badge.svg?branch=)](https://coveralls.io/github/testcontainers/testcontainers-dotnet?branch=master)
+## About 
+
+Testcontainers is a .NET library  that helps to develop test automation - specially integration test scenarios. Under the hood, it uses  XUnit  framework and docker containers. https://testcontainers.org
```
Comment (isen-ng) [inline README.md:6]: Same over here. Putting in a newline in the text will not put a new line in the rendered markdown. Please keep lines to 120 characters for readability.

```suggestion
Testcontainers is a .NET library  that helps to develop test automation - specially integration test scenarios. Under 
the hood, it uses  XUnit  framework and docker containers. https://testcontainers.org
```

File: README.md [hunk 3]
```diff
@@ -1,6 +1,138 @@
...
+
+Testcontainers make the following kinds of tests easier:
+
+- **Data access layer integration tests**: use a containerised instance of a MySQL, PostgreSQL or Oracle database to test your data access layer code for complete compatibility, but without requiring complex setup on developers' machines and safe in the knowledge that your tests will always start with a known DB state. Any other database type that can be containerised can also be used.
```
Comment (isen-ng) [inline README.md:10]: Split into new lines of 120 characters or less
```suggestion
- **Data access layer integration tests**: use a containerised instance of a MySQL, PostgreSQL or Oracle database to 
  test your data access layer code for complete compatibility, but without requiring complex setup on developers' 
  machines and safe in the knowledge that your tests will always start with a known DB state. Any other database type 
  that can be containerised can also be used.
```

File: README.md [hunk 4]
```diff
@@ -1,6 +1,138 @@
...
+- **Application integration tests**: for running your application in a short-lived test mode with dependencies, such as databases, message queues or web servers.
```
Comment (isen-ng) [inline README.md:11]:
```suggestion
- **Application integration tests**: for running your application in a short-lived test mode with dependencies, such 
  as databases, message queues or web servers.
```

File: README.md [hunk 5]
```diff
@@ -1,6 +1,138 @@
...
+- **UI/Acceptance tests**: use containerised web browsers, compatible with Selenium, for conducting automated UI tests. Each test can get a fresh instance of the browser, with no browser state, plugin variations or automated browser upgrades to worry about. And you get a video recording of each test session, or just each session where tests failed.
```
Comment (isen-ng) [inline README.md:12]:
```suggestion
- **UI/Acceptance tests**: use containerised web browsers, compatible with Selenium, for conducting automated UI 
  tests. Each test can get a fresh instance of the browser, with no browser state, plugin variations or automated 
  browser upgrades to worry about. And you get a video recording of each test session, or just each session where 
  tests failed.
```

File: README.md [hunk 6]
```diff
@@ -1,6 +1,138 @@
...
+
+```
+Install-Package TestContainers
+```
+
+Now, you can begin writing your integration test with throwaway docker containers.
+
+### Operating Systems
+
+Testcontainers.NET supports Windows, Linux, and macOS as host systems. Linux Docker containers are supported on all three operating systems.
```
Comment (isen-ng) [inline README.md:35]: Split into new lines
```suggestion
Testcontainers.NET supports Windows, Linux, and macOS as host systems. Linux Docker containers are supported on all 
three operating systems.
```

File: README.md [hunk 7]
```diff
@@ -1,6 +1,138 @@
...
+
+Native Windows Docker containers are only supported on Windows. Windows requires the host operating system version to match the container operating system version. You'll find further information about Windows container version compatibility [here](https://docs.microsoft.com/en-us/virtualization/windowscontainers/deploy-containers/version-compatibility).
```
Comment (isen-ng) [inline README.md:37]:
```suggestion
Native Windows Docker containers are only supported on Windows. Windows requires the host operating system version to 
match the container operating system version. You'll find further information about Windows container version 
compatibility [here](https://docs.microsoft.com/en-us/virtualization/windowscontainers/deploy-containers/version-compatibility).
```

File: README.md [hunk 8]
```diff
@@ -1,6 +1,138 @@
...
+
+Keep in mind to enable the correct Docker engine on Windows host systems to match the container operating system. With Docker CE you can switch the engine with: `$env:ProgramFiles\Docker\Docker\DockerCli.exe -SwitchDaemon` or `-SwitchLinuxEngine`, `-SwitchWindowsEngine`.
```
Comment (isen-ng) [inline README.md:39]:
```suggestion
Keep in mind to enable the correct Docker engine on Windows host systems to match the container operating system. With 
Docker CE you can switch the engine with: `$env:ProgramFiles\Docker\Docker\DockerCli.exe -SwitchDaemon` or 
`-SwitchLinuxEngine`, `-SwitchWindowsEngine`.
```

File: README.md [hunk 9]
```diff
@@ -1,6 +1,138 @@
...
+- `WithMount` mounts a volume into the container e. g. `-v, --volume .:/tmp`.
+- `WithCleanUp` removes a stopped container automatically.
+- `WithOutputConsumer` redirects `stdout` and `stderr` to capture the Testcontainer output.
+- `WithWaitStrategy` sets the wait strategy to complete the Testcontainer start and indicates when it is ready.
+- `WithDockerfileDirectory` builds a Docker image based on a Dockerfile (`ImageFromDockerfileBuilder`).
+- `WithDeleteIfExists` removes the Docker image before it is rebuilt (`ImageFromDockerfileBuilder`).
+
+## Examples
+
+Here is an example of a pre-configured Testcontainer. In the example, Testcontainers starts a PostgreSQL database and executes a SQL query.
```
Comment (isen-ng) [inline README.md:84]: split

File: README.md [hunk 10]
```diff
@@ -1,6 +1,138 @@
...
+            dbConnection.Close();
+            await container.Stop();
+        }
+```
+
+
+
+## Contributing
+
+Please read [CONTRIBUTING.md](CONTRIBUTING.md) for details on our code of conduct, and the process for submitting pull requests to us.
```
Comment (isen-ng) [inline README.md:114]: split

File: README.md [hunk 11]
```diff
@@ -1,6 +1,138 @@
...
+We use [SemVer](http://semver.org/) for versioning. For the versions available, see the [tags on this repository](https://github.com/testcontainers/testcontainers-dotnet/tags). 
+
+## Authors
+
+* **Gurpreet Singh Sohal** - [swissarmykirpan](https://github.com/swissarmykirpan)
+* **ISen** - [isen-ng](https://github.com/isen-ng)
+* **Andre Hofmeister** - [HofmeisterAn](https://github.com/HofmeisterAn)
+* **Lalit Kale** - [lalitkale](https://github.com/lalitkale)
+
+See also the list of [contributors](https://github.com/testcontainers/testcontainers-dotnet/graphs/contributors) who participated in this project.
```
Comment (isen-ng) [inline README.md:127]: split

File: README.md [hunk 12]
```diff
@@ -1,6 +1,138 @@
...
+
+
+
+## Contributing
+
+Please read [CONTRIBUTING.md](CONTRIBUTING.md) for details on our code of conduct, and the process for submitting pull requests to us.
+
+## Versioning
+
+We use [SemVer](http://semver.org/) for versioning. For the versions available, see the [tags on this repository](https://github.com/testcontainers/testcontainers-dotnet/tags).
```
Comment (isen-ng) [inline README.md:118]: split

**Review by isen-ng — CHANGES_REQUESTED** [review]
One of the purposes of Markdown was to allow writing text that is easily readable without rendering (on a computer screen). Thus the markdown we write should be easily readable without rendering as well. IMO that means no horizontal scrolling, well-inserted new lines, etc.

**Comment by isen-ng** [comment]
I'm going to review the formatting first. Once that's settled then I'll move on to the content.

Thank you for the PR @lalitkale

**Comment by lalitkale** [comment]
@isen-ng thanks for your feedback. This is what makes open source great. I will work on the comments and will update the PR.

File: README.md [hunk 13]
```diff
@@ -1,6 +1,138 @@
...
+- [ ] Startup strategy
+- [ ] Wait Strategy
+- [ ] Accessing container Logs
+- [ ] Creating Docker Images on the fly
+- [ ] Custom configuration - overwrite some default properties of environments
+- [ ] Advanced Options
+- [ ] out of the box support for different technologies, databases and webdrivers through modules extension
+- [ ] Enhancement of Unit Testing Framework Support from current XUnit Framework
+
+## Commands
```
Comment (HofmeisterAn) [inline README.md:62]: I don't think all of these commands are supported yet by the official version. This is the list of my implementation.

File: README.md (hunk 7 above)
Comment (HofmeisterAn) [inline README.md:37]: We don't have any native Windows container yet.

File: README.md (hunk 13 above)
Comment (lalitkale) [reply README.md:62]: yes, I reffered your implementation and readme for these commands. I was under assumption that after merging all three versions, we will be providing the same soon. Never mind, will revisit them with current implementation.

File: README.md (hunk 7 above)
Comment (lalitkale) [reply README.md:37]: okay, will remove it.

File: README.md (hunk 13 above)
Comment (HofmeisterAn) [reply README.md:62]: I'm not sure if they match the official names. I would wait until we have taken over the functions, we will probably take more from @isen-ng, idk. Anyway, thanks for the PR.

**Comment by jzabroski** [comment]
This is a really useful README.md. what is left to merge it??

**Comment by isen-ng** [comment]
@jzabroski We have left comments for the author but it seems that he hasn't implemented the changes.

Please reopen if there is any intention to get this PR back to latest and comments implemented.

**Comment by jzabroski** [comment]
That's lame. Why not accept it and fix it yourself? There's so much useful information here. Nobody will want to get involved with this project with its current README.md

**Comment by jzabroski** [comment]
I guess the other odd thing is your personal master branch has these helpful docs but official project github doesn't. :(

**Comment by isen-ng** [comment]
> That's lame. Why not accept it and fix it yourself? There's so much useful information here. Nobody will want to get involved with this project with its current README.md

Unfortunately, this is how it is. We do not want to copy his work and commit it using our own accounts, nor do want want to commit an incomplete PR, which will remain on his commit history forever like a stain.

> I guess the other odd thing is your personal master branch has these helpful docs but official project github doesn't. :(

My personal repository is something different from this. I hold full ownership of that repository and that is why I could advance it further without repercussions. On this organizational repository, maintainers must work as a team to discuss and agree on a PR/proposal before being committed. Sometimes, unfortunately, this takes a much longer time due to disagreements and conflicting schedules.

P.S. the APIs on my repository is completely different from this repository's in it's current form and you will not be able to simply switch it out.

**Comment by jzabroski** [comment]
Is your version hosted on Nuget.org?  It seems like your version and TestEnvironment.Docker are the state of the art; I'm leaning towards trying TestEnvironment.Docker first, as it has fluent async/await syntax which seems more flexible for testing scenarios like bringing up 10,000 nodes concurrently and having some of them die.

**Pull request closed without merging**